src/main/splitting/splitting.py

```diff
@@ -1,117 +1,76 @@
-
-    for curr_split in supposed_splits:
-        curr_intersected_tasks = intersect(prev_intersected_tasks, curr_split[consts.SPLIT_DICT.TASKS.value])
-        if len(curr_intersected_tasks) == 0:
-            # it means that the supposed task has changed, so we should split on prev_split
-            append_new_split_to_list(real_splits, prev_split[consts.SPLIT_DICT.INDEX.value],
-                                     prev_split[consts.SPLIT_DICT.RATE.value], prev_intersected_tasks)
-            prev_intersected_tasks = curr_split[consts.SPLIT_DICT.TASKS.value]
-        else:
-            prev_intersected_tasks = curr_intersected_tasks
-
-        prev_split = curr_split
-
-    # add the last split
-    append_new_split_to_list(real_splits, prev_split[consts.SPLIT_DICT.INDEX.value],
-                             prev_split[consts.SPLIT_DICT.RATE.value], prev_intersected_tasks)
-    return real_splits
-
-
-def get_file_and_parent_folder_names(file: str):
-    return os.path.join(get_parent_folder_name(file), get_name_from_path(file))
-
-
-def filter_already_tested_files(files: list, result_folder_path: str):
-    tested_files = get_all_file_system_items(result_folder_path, ct_file_condition, consts.FILE_SYSTEM_ITEM.FILE.value)
-    # to get something like 'ati_239/Main_2323434_343434.csv'
-    tested_folder_and_file_names = list(
-        map(lambda f: get_file_and_parent_folder_names(f), tested_files))
-    return list(filter(lambda f: get_file_and_parent_folder_names(f) not in tested_folder_and_file_names, files))
-
-
-def run_tests(path: str):
-    log.info(f'Start running tests on path {path}')
-    result_folder = get_result_folder(path, consts.RUNNING_TESTS_RESULT_FOLDER)
-
-    files = get_all_file_system_items(path, ct_file_condition, consts.FILE_SYSTEM_ITEM.FILE.value)
-    str_len_files = str(len(files))
-    log.info(f'Found {str_len_files} files to run tests on them')
-
-    files = filter_already_tested_files(files, result_folder)
-    str_len_files = str(len(files))
-    log.info(f'Found {str_len_files} files to run tests on them after filtering already tested')
-
-    tasks = [t.value for t in consts.TASK]
-    in_and_out_files_dict = create_in_and_out_dict(tasks)
-
-    for i, file in enumerate(files):
-        file_log_info = f'file: {str(i + 1)}/{str_len_files}'
-        log.info(f'Start running tests on {file_log_info}, {file}')
-        data = pd.read_csv(file, encoding=consts.ISO_ENCODING)
-        language, data = check_tasks_on_correct_fragments(data, tasks, in_and_out_files_dict, file_log_info)
-        log.info(f'Finish running tests on {file_log_info}, {file}')
-        write_based_on_language(result_folder, file, data, language)
-
-    return result_folder
+        log.error(f'Several tasks are solved: {solved_tasks}, tests results: {tests_results}')
+        raise ValueError(f'Several tasks are solved: {solved_tasks}, tests results: {tests_results}')
```

Comment:
Do we really want to throw an error here? It will stop handler for all files. Maybe it is better to use some special value for this case?

Reply:
Yes, it will, but I thought that having two tasks solved is an incorrect case, meaning that we have a mistake in run_tests function or anywhere else. So, we cannot continue data handling anyway. Having some special value for this case instead of error throwing forces us to do an additional check for this value after splitting and stop everything in case of a successful check. 
However, this process of splitting takes not so many time to be worried about stopping, and I think that throwing an error is the best way to know about errors asap.